File: server/Services/PeopleManagement/PeopleManagement.Domain/AggregatesModel/RoleAggregate/Department.cs
namespace PeopleManagement.Domain.AggregatesModel.RoleAggregate
{
    public sealed class Department : Entity
    {
        public Name Name { get; private set; } = null!;
        public Description Description { get; private set; } = null!;

        private Department() { }
        private Department(Name name, Description description)
        {
            Name = name;
            Description = description;
        }

        public static Department Create(Name name, Description description) => new(name, description);
    }
}

File: client/rufino/lib/modules/employee/domain/role_info/department.dart
import 'package:rufino/modules/employee/domain/model/base/enumeration.dart';
import 'package:rufino/modules/employee/domain/model/base/text_base.dart';
import 'package:rufino/modules/employee/domain/role_info/department_description.dart';

class Department extends Enumeration {
  final DepartmentDescription depertmentDescription;

  const Department(String id, String name, this.depertmentDescription)
      : super(id, name, "Setor");
  const Department.empty(
      {this.depertmentDescription = const DepartmentDescription.empty()})
      : super.empty(displayName: "Setor");

  static Department fromJson(Map<String, dynamic> json) {
    return Department(
        json["id"], json["name"], DepartmentDescription(json["description"]));
  }

  static List<Department> fromListJson(List<dynamic> listJson) {
    return listJson.map((el) => fromJson(el)).toList();
  }

  List<TextBase> get textBase => [depertmentDescription];
}
